### PR TITLE
[ethercat] Ethercat checks for required slave count

### PR DESCRIPTION
**Pull request by magni-mar**

### CODE REVIEW

File: exes/ethercat/inc/public/tfc/ec/config/bus.hpp [hunk 1]
```diff
@@ -39,11 +40,15 @@ struct tfc::json::detail::to_json_schema<tfc::ec::config::network_interface> {
 };
 
 namespace tfc::ec::config {
-struct bus {
+struct ethercat {
   network_interface primary_interface{ tfc::global::get_interfaces()[0] };
+  confman::observable<std::optional<std::size_t>> minimum_slave_count{ 0 };
```
Comment (omarhogni) [inline exes/ethercat/inc/public/tfc/ec/config/bus.hpp:45]:
```suggestion
  confman::observable<std::optional<std::size_t>> required_slave_count{ 0 };
```

I don't see a use case for the minimum amount. Should we not just exact match?

File: exes/ethercat/inc/public/tfc/ec.hpp [hunk 2]
```diff
@@ -164,6 +164,16 @@ class context_t {
...
+    if (config_->value().minimum_slave_count.value().has_value()) {
+      if (slave_count() < config_->value().minimum_slave_count.value().value()) {
+        logger_.trace("Minimum slave count not reached, current slave count: {}, minimum slave count: {}", slave_count(),
+                      config_->value().minimum_slave_count.value().value());
+        return async_start();
+      }
+      logger_.trace("Slave count reached, current slave count: {}, minimum slave count: {}", slave_count(),
+                    config_->value().minimum_slave_count.value().value());
+    }
+
```
Comment (omarhogni) [inline exes/ethercat/inc/public/tfc/ec.hpp:176]: Could you clean this up a little bit? maybe take a temporary and reference that in the if statements. This is allot of noise to functionality.

**Review by omarhogni — CHANGES_REQUESTED** [review]

File: exes/ethercat/inc/public/tfc/ec/config/bus.hpp (hunk 1 above)
Comment (magni-mar) [reply exes/ethercat/inc/public/tfc/ec/config/bus.hpp:45]: sounds good, changing

File: exes/ethercat/inc/public/tfc/ec.hpp (hunk 2 above)
Comment (magni-mar) [reply exes/ethercat/inc/public/tfc/ec.hpp]: added variable.

File: exes/ethercat/inc/public/tfc/ec.hpp [hunk 3]
```diff
@@ -164,6 +164,18 @@ class context_t {
       return async_start();
     }
 
+    std::optional<std::size_t> configured_slave_count = config_->value().required_slave_count.value();
```
Comment (omarhogni) [inline exes/ethercat/inc/public/tfc/ec.hpp:167]:
```suggestion
    auto configured_slave_count = config_->value().required_slave_count.value();
```

File: exes/ethercat/inc/public/tfc/ec.hpp [hunk 4]
```diff
@@ -164,6 +164,18 @@ class context_t {
...
+
+    if (configured_slave_count.has_value()) {
+      if (slave_count() == configured_slave_count.value()) {
+        logger_.trace("Slave count is wrong, current slave count: {}, required slave count: {}", slave_count(),
```
Comment (omarhogni) [inline exes/ethercat/inc/public/tfc/ec.hpp:171]:
```suggestion
        logger_.error("Slave count is wrong, current slave count: {}, required slave count: {}", slave_count(),
```

File: exes/ethercat/inc/public/tfc/ec/config/bus.hpp [hunk 5]
```diff
@@ -39,11 +40,15 @@ struct tfc::json::detail::to_json_schema<tfc::ec::config::network_interface> {
 };
 
 namespace tfc::ec::config {
-struct bus {
+struct ethercat {
   network_interface primary_interface{ tfc::global::get_interfaces()[0] };
+  confman::observable<std::optional<std::size_t>> required_slave_count{ 0 };
```
Comment (omarhogni) [inline exes/ethercat/inc/public/tfc/ec/config/bus.hpp:45]:
```suggestion
  confman::observable<std::optional<std::size_t>> required_slave_count{ std::nullopt };
```

**Review by omarhogni — CHANGES_REQUESTED** [review]

**Review by omarhogni — APPROVED** [review]

**Pull request merged**